From beb34424107f6b09271dc468c045de14d4d40ec9 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Fri, 18 Jun 2021 10:47:19 +0530
Subject: [PATCH 1591/1921] octeontx2-af: Fix writing to incorrect register
 offset

If more than one scheduler queue is allocated to
a PF then bogus register offset is written
because of negative lbk_links variable. Fix it
by initializing lbk_links variable before the loop.

Fixes: 9b7eb6e5076c ("octeontx2-af: Change transmit side NPC rules")
Change-Id: I0c155479dad07228d4489bc538fb07fc57af7f3b
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/54333
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
index 8dc29af02a82..d11afe1e63a9 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
@@ -2257,7 +2257,6 @@ static void rvu_nix_tx_tl2_cfg(struct rvu *rvu, int blkaddr,
 		return;
 
 	lbk_link_start = hw->cgx_links;
-	lbk_links = hw->lbk_links;
 
 	for (schq = 0; schq < txsch->schq.max; schq++) {
 		if (TXSCH_MAP_FUNC(txsch->pfvf_map[schq]) != pcifunc)
@@ -2265,6 +2264,7 @@ static void rvu_nix_tx_tl2_cfg(struct rvu *rvu, int blkaddr,
 		/* Enable all LBK links with channel 63 by default so that
 		 * packets can be sent to LBK with a NPC TX MCAM rule
 		 */
+		lbk_links = hw->lbk_links;
 		while (lbk_links--)
 			rvu_write64(rvu, blkaddr,
 				    NIX_AF_TL3_TL2X_LINKX_CFG(schq,
-- 
2.31.1

